creator-node/src/snapbackSM.js

```diff
@@ -47,63 +58,308 @@ class SnapbackSM {
+
+    // Issue sync request to secondary
+    await axios(syncRequestParameters)
+
+    // Monitor the sync status
+    await this.monitorSecondarySync(syncWallet, primaryClockValue, secondaryUrl)
+
+    // Exit when sync status is computed
+    // Determine how many times to retry this operation
+    this.log('------------------END Process SYNC------------------')
+  }
+
+  // Function which ensures that state machine has been initialized correctly
+  // If not available on startup, subsequent state machine will attempt to initialize until success
+  async initializeNodeIdentityConfig () {
+    this.log(`Initializing SnapbackSM`)
+    this.log(`Retrieving spID for ${this.endpoint}`)
+    const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
+      this.endpoint
+    )
+    // A returned spID of 0 means this this.endpoint is currently not registered on chain
+    // In this case, the stateMachine is considered to be 'uninitialized'
+    if (recoveredSpID === 0) {
+      this.initialized = false
+    }
+    config.set('spID', recoveredSpID)
+    this.log(`Recovered ${config.get('spID')} for ${this.endpoint}`)
+    this.initialized = true
+    return this.initialized
+  }
+
+  // Initialize the state machine
   async init () {
-    const endpoint = config.get('creatorNodeEndpoint')
+    await this.stateMachineQueue.empty()
+    await this.syncQueue.empty()
+
     const isUserMetadata = config.get('isUserMetadataNode')
     if (isUserMetadata) {
-      this.log(`SnapbackSM disabled for userMetadataNode. ${endpoint}, isUserMetadata=${isUserMetadata}`)
+      this.log(`SnapbackSM disabled for userMetadataNode. ${this.endpoint}, isUserMetadata=${isUserMetadata}`)
       return
     }
 
-    if (!this.initialized) {
-      this.log(`Initializing SnapbackSM`)
-      this.log(`Retrieving spID for ${endpoint}`)
-      const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
-        endpoint
-      )
-      // A returned spID of 0 means this endpoint is currently not registered on chain
-      // In this case, the stateMachine is considered to be 'uninitialized'
-      if (recoveredSpID === 0) return
-      config.set('spID', recoveredSpID)
-      this.initialized = true
-    }
-    this.log(`Recovered ${config.get('spID')} for ${endpoint}`)
+    // Initialize state machine queue processor
+    this.stateMachineQueue.process(
+      async (job, done) => {
+        try {
+          await this.processStateMachineOperation(job)
+        } catch (e) {
+          this.log(`stateMachineQueue error processing ${e}`)
+        } finally {
+          // TODO: Remove dev mode
+          this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
+          await utils.timeout(DevDelayInMS)
+          this.stateMachineQueue.add({ startTime: Date.now() })
+          done()
```

Comment:
@dmanjunath @SidSethi how should we determine this sync interval?

Reply:
can we make the delay an envvar to easily tune after deploy?

Reply:
no, this will never be run with dev mode enabled - we will exclusively be on cron. so 'tuning' of a dev var like this would add 0 value. resolving